### PR TITLE
support dictproxy views Fixes #295

### LINKED ISSUE #295: AttributeError: 'dictproxy' object has no attribute 'viewitems', 'viewvalues', 'viewkeys'
```
Python 2.7.16 (default, Apr  6 2019, 01:42:57) 
[GCC 8.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> class Ham(object):
...     pass
... 
>>> import six
>>> six.viewitems(vars(Ham))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'dictproxy' object has no attribute 'viewitems'
>>> six.viewvalues(vars(Ham))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'dictproxy' object has no attribute 'viewvalues'
>>> six.viewkeys(vars(Ham))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'dictproxy' object has no attribute 'viewkeys'
```

strictly speaking this might be a cPython bug, but I think the six methods need to include a check for dictproxy and handle this case

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: test_six.py [hunk 1]
```diff
@@ -434,6 +434,32 @@ def stock_method_name(viewwhat):
...
+
+    dictproxy = vars(Ham)
+    dictcopy = dict(dp)
+
+    keysview = six.viewkeys(dictproxy)
+    valuesview = six.viewvalues(dictproxy)
+    itemsview = six.viewitems(dictproxy)
+
+    # test dictproxy six.view*s work the same as per a regular dict
+    assert keysview == six.viewvalues(dictcopy)
```
Comment (graingert) [inline test_six.py]:
```suggestion
    assert keysview == six.viewkeys(dictcopy)
```

**Comment by graingert** [comment]
@benjaminp I think this fixes the issues with my previous attempt

**Comment by graingert** [comment]
@benjaminp any thoughts on this?

**Comment by graingert** [comment]
https://github.com/benjaminp/six/issues/295

**Comment by graingert** [comment]
@benjaminp any thoughts on this?

**Comment by benjaminp** [comment]
Thank you for the PR. As I said in https://github.com/benjaminp/six/pull/296#issuecomment-546179264, I don't want to complicate `view*` for this, especially since there's no way to view dict proxy's on Python 2.

**Comment by graingert** [comment]
> Thank you for the PR. As I said in [#296 (comment)](https://github.com/benjaminp/six/pull/296#issuecomment-546179264), I don't want to complicate `view*` for this, especially since there's no way to view dict proxy's on Python 2.

But it does work for dictproxy, you just use the `collections.*View`

**Comment by benjaminp** [comment]
Sorry, I meant a method on the type.

**Comment by graingert** [comment]
I'd like you to reconsider, because it makes it much easier to use six.viewitems everywhere it would also be supported in Py3 and much easier to produce safe automated transforms when upgrading a py2 codebase to py3

**Comment by benjaminp** [comment]
> I'd like you to reconsider,

I've already considered it twice.

> because it makes it much easier to use six.viewitems everywhere it would also be supported in Py3 (eg `class Foo(object): ...; six.viewitems(vars(Foo))` and much easier to produce safe automated transforms when upgrading a py2 codebase to py3

What automated transform? If the Python 2 code is doing `something.viewitems()`, then it already doesn't work it dict proxies. If it's doing `something.iteritems()`, `six.iteritems(something)` will work fine.

**Pull request closed without merging**